More efficient mechanism for fetching/batching orders in the control panel

alecpm opened the pull request:
The prior implementation retrieved all orders and then sorted them before filtering them by date. This implementation retrieves only the keys (which include a date), filters and sorts those keys, and then provides a lazy list implementation for retrieving the order items which can be used for batching.

This will also require some changes to [jazkarta.pfg.jazshop](https://github.com/jazkarta/jazkarta.pfg.jazshop) which imports and uses the private `_fetch_orders()` function.

sauzher committed:
use of plone mockup pickadate

sauzher committed:
fix #70
update changelog

alecpm committed:
Implement orders control panel performance improvements for date filtering and bactching.

alecpm commented:
This also incorporates the pick-a-date date picker switch from issue #70 (PR #75)

alecpm commented:
See jazkarta/jazkarta.pfg.jazshop#2 for related updates to that package.

fulv reviewed (changes requested):
I get a 404 when I try loading @@jazkarta-shop-orders now.

alecpm commented:
@fulv Is that happening on your local instance when using this branch?

alecpm committed:
Allow for empty control panel date selection.

alecpm committed:
Apparently __name__ isn't always available on NCI views?

alecpm commented:
@fulv I've fixed the issue causing the 404. For some reason on NCI `view/__name__` was throwing an attribute error. That's not a change that I introduced, but it may have been added on master at some point. It was also breaking the shipping control panel, but it should be working now.

I've deployed the changes to staging and it seems to be performing quite well there.

alecpm commented:
Oddly, I didn't see that `view/__name__` issue when testing on my local copy of NCI staging (just pulled this morning), but it did happen on staging. Very strange.

alecpm committed:
Better decimal handling in order display.

alecpm committed:
Fix CSV export.

alecpm committed:
Fixes for CSV export.

jessesnyder commented:
Order filtering & speed, and CSV download all working well for me now.

jessesnyder committed:
Standard EMAIL_RE is already anchored to start of target string

jessesnyder committed:
More broadly compatible version of EMAIL_RE test
